xdsl/dialects/riscv.py

```diff
@@ -2809,11 +2810,19 @@ class RegisterAllocatedMemoryEffect(MemoryEffect):
     """
 
     @classmethod
-    def has_effects(cls, op: Operation) -> bool:
-        return any(
+    def get_effects(cls, op: Operation) -> set[MemoryEffectKind]:
```

Comment:
I am wondering if it makes sense to have MemoryEffectKind be a [Flag](https://docs.python.org/3/library/enum.html#enum.Flag) instead?

Reply:
No I don't think it does, this kind of effect is one aspect of an effect, a next stage equips them with different resources or effected values to make up an effect and I don't think having a Flag composed in there would make anything simpler.

Otherwise, similar comment to other comments from Sasha; I'm just partially porting stuff from MLIR here; I'd rather have it as similar as possible at least until it's on par; I don't see the value of nitting on those details at the cost of having the port be error-prone on the behaviour first.

Reply:
Same addition as on sasha's discussion, will mention these points inline in places